pyomo/contrib/cp/transform/logical_to_disjunctive_walker.py

```diff
@@ -243,6 +243,13 @@ def initializeWalker(self, expr):
         return True, expr
 
     def beforeChild(self, node, child, child_idx):
+        if child.__class__ is bool:
+            # If we encounter a bool, we are going to need to treat it as
+            # binary explicitly because we are finally pedantic enough in the
+            # expression system to not allow some of the mixing we will need
+            # (like summing a LinearExpression with a bool)
+            return False, int(child)
+
         if child.__class__ in EXPR.native_types:
             return False, child
```

Comment:
This is fine, however two questions:
1. Would the following be more efficient:
```python
if child.__class__ in EXPR.native_types:
    if child.__class__ is bool:
        # If we encounter a bool, we are going to need to treat it as
        # binary explicitly because we are finally pedantic enough in the
        # expression system to not allow some of the mixing we will need
        # (like summing a LinearExpression with a bool)
        return False, int(child)
    return False, child
```
2. are there concerns with float constants here?  For Param, we error if the value is fractional, but not here.  If the same concern applies, then maybe we should just do:
```python
if child.__class__ in EXPR.native_types:
    val = int(child)
    if val != child:
        raise ValueError("...")
    return False, val
```
(which covers both the fractional values and the bool case)

Reply:
Oh, good catch! But I think it's actually the reverse: I shouldn't check in Param or here because it won't be required when we probably someday allow relational expressions to be in logical expressions here. And we are checking for integrality in places where we need it (e.g., `atmost`, `atleast`, and `exactly`)